batchconsumer/batchermanager.go

```diff
@@ -0,0 +1,208 @@
+package batchconsumer
+
+import (
+	"os"
+	"time"
+
+	kv "gopkg.in/Clever/kayvee-go.v6/logger"
+
+	"github.com/Clever/amazon-kinesis-client-go/batchconsumer/stats"
+	"github.com/Clever/amazon-kinesis-client-go/kcl"
+)
+
+type tagMsgPair struct {
+	tag  string
+	msg  []byte
+	pair kcl.SequencePair
+}
+
+type batcherManager struct {
+	log           kv.KayveeLogger
+	sender        Sender
+	chkpntManager *checkpointManager
+
+	batchCount    int
+	batchSize     int
+	batchInterval time.Duration
+
+	batchMsg      chan tagMsgPair
+	lastIgnored   chan kcl.SequencePair
+	lastProcessed chan kcl.SequencePair
+	shutdown      chan chan<- struct{}
+}
+
+func NewBatcherManager(
+	sender Sender, chkpntManager *checkpointManager, config Config, log kv.KayveeLogger,
+) *batcherManager {
+	bm := &batcherManager{
+		log:           log,
+		sender:        sender,
+		chkpntManager: chkpntManager,
+
+		batchCount:    config.BatchCount,
+		batchSize:     config.BatchSize,
+		batchInterval: config.BatchInterval,
```

Comment:
nit but IMO slightly clearer to take in these config values individually vs the bigger Config object with stuff the batchermanager doesn't use